generators/entity/templates/src/main/java/package/service/impl/_EntityServiceImpl.java

```diff
@@ -95,8 +95,14 @@ public void delete(<%= pkType %> id) {
      *  @param query the query of the search
      *  @return the list of entities
      */<% if (databaseType == 'sql') { %>
-    @Transactional(readOnly = true) <% } %>
-    public List<<%= instanceType %>> search(String query) {
-        <%- include('../../common/search_template', {viaService: viaService}); -%>
+    @Transactional(readOnly = true)<% } %>
```

Comment:
If it helps maintenance then yes we can split
On 22 Feb 2016 10:02, "Gerald Humphries" <notifications@github.com> wrote:

> In
> generators/entity/templates/src/main/java/package/service/impl/_EntityServiceImpl.java
> <https://github.com/jhipster/generator-jhipster/pull/2953#discussion_r53580321>
> :
>
> > @@ -95,8 +95,14 @@ public void delete(<%= pkType %> id) {
> >       *  @param query the query of the search
> >       *  @return the list of entities
> >       */<% if (databaseType == 'sql') { %>
> > -    @Transactional(readOnly = true) <% } %>
> > -    public List<<%= instanceType %>> search(String query) {
> > -        <%- include('../../common/search_template', {viaService: viaService}); -%>
> > +    @Transactional(readOnly = true)<% } %>
>
> I can do that but I'll need to split up search_template. Should we
> refactor get_all_template too while we're at it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jhipster/generator-jhipster/pull/2953/files#r53580321>
> .
>